rctv/main.py

```diff
@@ -92,3 +110,65 @@ async def rctv(
             "rc_payload": rc_payload,
         },
     )
+
+
+@app.get("/api/events")
+async def cal(
+    _request: Request,
+    response: Response,
+    background_tasks: BackgroundTasks,
+    start: str | None = None,
+    end: str | None = None,
+):
+    """
+    Takes an optional query paramters ?start=YYYY-MM-DD and &end=YYYY-MM-DD
+    Returns calendar events as JSON between [start, end] date ranges
+    """
+
+    # This either returns the cached calendar or makes an API call
+    # This isn't ideal because if some RCTV is not regularly hitting this API we
+    # are going to get cache-misses and do the full API request which comes out
+    # to be 1.2 MB
+    #
+    # An alternative approach is to use something like fastapi_utils.repeat_every
+    # to run this on a periodic basis instead:
+    #
+    # https://fastapi-utils.davidmontague.xyz/user-guide/repeated-tasks/
```

Comment:
@gregsadetsky If you know how to do things on a background loop in Django then try it. Otherwise this kind of works?